taskiq_nats/broker.py

```diff
@@ -80,37 +88,57 @@ async def shutdown(self) -> None:
         await super().shutdown()
 
 
-class JetStreamBroker(AsyncBroker):  # noqa: WPS230
-    """
-    JetStream broker for taskiq.
+class BaseJetStreamBroker(  # noqa: WPS230 (too many attrs)
+    AsyncBroker,
+    ABC,
+    typing.Generic[JetStreamConsumerType],
+):
+    """Base JetStream broker for taskiq.
 
-    This broker creates a JetStream context
-    and uses it to send and receive messages.
+    It has two subclasses - PullBasedJetStreamBroker
+    and PushBasedJetStreamBroker.
+
+    These brokers create a JetStream context
+    and use it to send and receive messages.
 
     This is useful for systems where you need to
     be sure that messages are delivered to the workers.
     """
 
     def __init__(  # noqa: WPS211 (too many args)
         self,
-        servers: Union[str, List[str]],
-        subject: str = "tasiq_tasks",
-        stream_name: str = "taskiq_jstream",
-        queue: Optional[str] = None,
-        result_backend: "Optional[AsyncResultBackend[_T]]" = None,
-        task_id_generator: Optional[Callable[[], str]] = None,
-        stream_config: Optional[StreamConfig] = None,
-        **connection_kwargs: Any,
+        servers: typing.Union[str, typing.List[str]],
+        subject: str = "taskiq_tasks",
+        stream_name: str = "taskiq_jetstream",
+        queue: typing.Optional[str] = None,
+        durable: str = "taskiq_durable",
+        result_backend: typing.Optional[AsyncResultBackend[_T]] = None,
```

Comment:
Can you remove `result_backend` and `task_id_generator` parameters. They are going to bedeprecated soon.

Reply:
Done

Reply:
@s3rius check plz